### ISSUE TITLE
Incorrect formatting behavior for redundantReturn rule

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#564]
We've added the `redudantReturn` rule to the Airbnb style guide (airbnb/swift#95). When we ran the new rule on our codebase, we noticed an issue where SwiftFormat behaved incorrectly.

### Example

```swift
var accessorThatReturnsOptionalEnum: SomeEnum? {
  // Notice that there's no space between () and {
  if let string = someMethodThatReturnsOptionalString(){
    return .value(string)
  } else {
    return nil
  }
}

enum SomeEnum {
  case value(String)
}

func someMethodThatReturnsOptionalString() -> String? {
  return nil
}
```

`$ swiftformat --rules redundantReturn .` reformats this by removing the `return`. This example no longer compiles:
```swift
var accessorThatReturnsOptionalEnum: SomeEnum? {
  // Notice that there's no space between () and {
  if let string = someMethodThatReturnsOptionalString(){
    .value(string) // 🛑 This `return` should not have been removed. This example no longer compiles.
  } else {
    return nil
  }
}

// ...
```

Notably, SwiftFormat behaves correctly if you add a space between the `()` and `{`:

```swift
var accessorThatReturnsOptionalEnum: SomeEnum? {
  // Space between () and {
  if let string = someMethodThatReturnsOptionalString() {
    return .value(string) // ✅ SwiftFormat doesn't make any changes
  } else {
    return nil
  }
}

// ...
```

**Comment by nicklockwood** [comment]
@calda thanks for reporting this, I'll get it fixed asap.

**Comment by nicklockwood** [comment]
@calda fixed in 0.44.3.
